persistence/sql/persister_jwk.go

```diff
@@ -96,7 +98,7 @@ func (p *Persister) UpdateKey(ctx context.Context, set string, key *jose.JSONWeb
 
 func (p *Persister) UpdateKeySet(ctx context.Context, set string, keySet *jose.JSONWebKeySet) error {
 	return p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		if err := p.DeleteKeySet(ctx, set); err != nil {
+		if err := p.DeleteKeySet(ctx, set); err != nil && errorsx.Cause(err) != x.ErrNotFound {
```

Comment:
This check is new. Did something regress? Also, please use `errors.Is()`

Reply:
See my comment below (https://github.com/ory/hydra/pull/3013#discussion_r838209491)